### PR TITLE
update(platform): byoc self-service GCP & AWS

### PR DESCRIPTION
**Pull request by wojcik-dorota**
https://aiven.atlassian.net/browse/MILK-7

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

### CODE REVIEW

File: docs/platform/howto/byoc/tag-custom-cloud-resources.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+import ConsoleLabel from "@site/src/components/ConsoleIcons"
+
+Custom cloud tags propagate to resources on the Aiven platform and in your own cloud infrastructure. Tagging allows resource categorization, which simplifies governance, cost allocation, and system performance review.
+
+## Types of tagging
+
+You can tag your custom cloud resources by:
+
+- [Service tagging](#service-tagging), affecting service nodes and VMs
```
Comment (jeremy-heller-aiven) [inline docs/platform/howto/byoc/tag-custom-cloud-resources.md:14]: Maybe change to "affecting Aiven service nodes and VMs"

File: docs/platform/howto/byoc/tag-custom-cloud-resources.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+- [Infrastructure tagging](#infrastructure-tagging), affecting all taggable infrastructure
```
Comment (jeremy-heller-aiven) [inline docs/platform/howto/byoc/tag-custom-cloud-resources.md:15]: Maybe update to "all taggable BYOC infrastructure"

File: docs/platform/concepts/byoc.md [hunk 3]
```diff
@@ -94,22 +102,24 @@ certain cases.
...
-The standard BYOC deployment requires you to create a Virtual Private Cloud
-(**BYOC VPC**) dedicated to Aiven-managed services within a cloud region you
-want to operate in. Aiven accesses this VPC from a static IP address and routes
+The AWS private deployment requires you to create a Virtual Private Cloud
+(**BYOC VPC**) dedicated to Aiven-managed services within a cloud region to operate in.
+Aiven accesses this VPC from a static IP address and routes
 traffic through a proxy for additional security. To accomplish this, Aiven
-utilizes a bastion host physically separated from the Aiven services you deploy.
-You are able to integrate your services using standard VPC peering techniques.
+utilizes a bastion host (**Bastion note**) physically separated from the Aiven services
```
Comment (jeremy-heller-aiven) [inline docs/platform/concepts/byoc.md:116]: minor misspelling: Bastion node

File: docs/platform/concepts/byoc.md [hunk 4]
```diff
@@ -94,22 +102,24 @@ certain cases.
...
+you deploy. The service VMs reside in privately addressed subnets (**Private Subnet**)
```
Comment (jeremy-heller-aiven) [inline docs/platform/concepts/byoc.md:117]: "reside in a privately addressed subnet"

File: docs/platform/concepts/byoc.md [hunk 5]
```diff
@@ -121,21 +131,94 @@ to work properly (supporting HA signaling to the Aiven management node and RPM d
...
 :::
 
+</TabItem>
+<TabItem value="2" label="AWS public">
+
+![AWS public model architecture diagram](/images/content/platform/concepts/byoc-aws-pub-arch.png)
+
+The AWS public deployment model requires you to create a Virtual Private Cloud (VPC)
+dedicated to Aiven-managed services within a cloud region to operate in (**BYOC VPC**).
+Aiven accesses this VPC through an Internet gateway. Service VMs reside in publicly
```
Comment (jeremy-heller-aiven) [inline docs/platform/concepts/byoc.md:141]: "in a publicly addressed subnet"

File: docs/platform/concepts/byoc.md [hunk 6]
```diff
@@ -121,21 +131,94 @@ to work properly (supporting HA signaling to the Aiven management node and RPM d
...
+The GCP private deployment model requires you to create two Virtual Private Clouds (VPCs)
+within a cloud region to operate in:
+
+-  VPC dedicated to Aiven-managed services (**Workload VPC**)
+-  VPC dedicated to a bastion host (**Bastion VPC**)
+
+Aiven accesses **Bastion VPC** from a static IP address and routes
+traffic through a proxy for additional security. To accomplish this, Aiven
+utilizes a bastion host (**Bastion note**) physically separated from the Aiven services
+you deploy. The service VMs reside in privately addressed subnets (**Private Subnet**)
```
Comment (jeremy-heller-aiven) [inline docs/platform/concepts/byoc.md:163]: "in a publicly addressed subnet"

File: docs/platform/concepts/byoc.md [hunk 7]
```diff
@@ -121,21 +131,94 @@ to work properly (supporting HA signaling to the Aiven management node and RPM d
...
+
+</TabItem>
+<TabItem value="4" label="GCP public">
+
+![GCP public model architecture diagram](/images/content/platform/concepts/byoc-gcp-pub-arch.png)
+
+The GCP public deployment requires you to create a Virtual Private Cloud (VPC)
+dedicated to Aiven-managed services within a cloud region to operate in (**Workload VPC**).
+Aiven accesses this VPC through an Internet gateway. Service VMs reside in publicly
+addressed subnet(s) (**Public Subnet**), and Aiven services can be accessed
```
Comment (jeremy-heller-aiven) [inline docs/platform/concepts/byoc.md:186]: "in a publicly addressed subnet"

**Review by jeremy-heller-aiven — DISMISSED** [review]

**Review by raphlopez — DISMISSED** [review]
A couple comments specific to the diagrams:
- [All] Since the security group rules we require from customers are both in ingress and egress categories, wouldn't it make more sense to have arrows be bidirectional (particularly the one from "Aiven Management" to "Bastion")? To me (and I'm no diagraming expect), a unidirectional arrow implies comms only travels in one direction. 
- [GCP] The concept of VPC is not bound to a region in GCP. Rather, a VPC is created at the project level, and subnets are defined in specific regions. Having VPC as a subcomponent of "Cloud Region" contradicts this.

**Comment by jeremy-heller-aiven** [comment]
@raphlopez I also think the arrows imply directionality, and we should have "Aiven Management" to "Bastion" with arrows going both directions.

With regard to GCP, we only offer BYOC within a single region on GCP (at least for now) so I suspect that's the reasoning behind the diagram. Maybe we replace "Cloud Region" with "GCP Project" and call out in the description that BYOC is only single-region. The message I would tell a customer is, "If you need another region, you need to create another BYOC".

**Comment by ArthurFlag** [comment]
@raphlopez The images have been updated.

**Pull request closed without merging**